File: backend/TeamDocumentServer/src/main/kotlin/ch/fhnw/woweb/TeamDocumentServer/service/JavaService.java
package ch.fhnw.woweb.TeamDocumentServer.service;

import reactor.core.publisher.Flux;

import org.reactivestreams.Publisher;
import reactor.core.publisher.FluxSink;

import java.util.List;
import java.util.function.Supplier;

public class JavaService {

    Flux<String> bridge = Flux.create(sink -> {
        new DocumentEventListener<String> (){
            @Override
            public void onDataChunk(List<String> chunk) {
                for(String s : chunk){
                    sink.next(s);
                }
            }

            @Override
            public void processComplete() {
                sink.complete();
            }
        };
    });

    public Flux<String> getFlux(){
        return bridge;
    }

    public void produceStream(List<String> elements){
        Flux.fromIterable(elements);
    }
}

File: backend/TeamDocumentServer/src/main/kotlin/ch/fhnw/woweb/TeamDocumentServer/service/JavaService.kt
package ch.fhnw.woweb.TeamDocumentServer.service

import org.springframework.stereotype.Service
import reactor.core.publisher.Flux
import reactor.core.publisher.FluxSink
import java.util.function.Consumer

@Service
class JavaService {
    var consumer: Consumer<List<String?>>? = null
    fun produceStream(): Flux<String?> {
        return Flux.create { sink: FluxSink<String?> ->
            consumer = Consumer { items: List<String?> ->
                items.forEach(
                    Consumer { t: String? -> sink.next(t!!) })
            }
        }
    }
}